BrowserKit/Sources/SiteImageView/SiteImageHandler.swift

```diff
@@ -15,12 +15,11 @@ public class DefaultSiteImageHandler: SiteImageHandler {
     private let urlHandler: FaviconURLHandler
     private let imageHandler: ImageHandler
 
-    private let serialQueue = DispatchQueue(label: "com.mozilla.DefaultSiteImageHandler")
-    private var _currentInFlightRequest: String?
-    private var currentInFlightRequest: String? {
-        get { return serialQueue.sync { _currentInFlightRequest } }
-        set { serialQueue.sync { _currentInFlightRequest = newValue } }
-    }
+    /// Right now, multiple `SiteImageView`s each have their own `DefaultSiteImageHandler`. Ideally they'd all share a reference to
+    /// the same `DefaultSiteImageHandler` so we could properly queue and throttle requests to get favicon URLs, download images,
+    /// etc. Since that's a large architectural change, for now lets use a static queue so we can prevent too many duplicate calls to remotely fetching
+    /// URLs and images. (FXIOS-9830, revised FXIOS-9427 bugfix)
+    private(set) static var requestQueue: [String: Task<UIImage, Never>] = [:]
```

Comment:
I think my only concern with this is the introduction of this global state; it's somewhat obscured that all of these instances will be potentially checking against the same shared request queue, even if they are initialized with different url handlers or image handlers in the initializer. Right now I don't think that matters much in practice, and it's hard to imagine how it would realistically cause problems but it might be something nice for us to revisit at some point.

Reply:
I totally understand. Ideally this would all be refactored to be using a single instance of `DefaultSiteImageHandler` passed in to each cell, but this would entail a big refactor...

I'm ok with leaving the current solution in as well, as long as we can get my new unit test to pass reliably for it. Since that (I believe) mimics the current state of affairs with multiple instances of DefaultSiteImageHandler being created for a single cell/site URL.

I'll experiment with it a bit this afternoon.

Reply:
We went back and forth on this a bunch when originally building it if this should be a singleton or not. In the end we opted to go with the simpler solution and live with the fact there would occasionally be the odd unnecessary request but obviously that has turned out to be a worse problem than we anticipated.

Reply:
I'm not too worried about the global state from an app stability perspective because it's private and it's within a pretty contained submodule but it's a little concerning from a test perspective because of the possibility of one test colliding with another. It's not a major concern tho and seems like a reasonable solution that limits the scope of the change. Just worth keeping an eye on if we start getting flaky tests in this area in the future we might want to revisit.

Reply:
Thanks @OrlaM! That makes sense. This is definitely something to keep an eye on in the future... I was browsing Sentry to see if I could find any `EXC_BAD_ACCESS` type crashes related to favicons and saw [this one](https://mozilla.sentry.io/issues/3942614988/events/062a35ac7340407bbd922d8d6c2a24ce/?project=6176941&query=is%3Aunresolved+issue.priority%3A%5Bhigh%2C+medium%5D&referrer=previous-event&statsPeriod=14d&stream_index=4). Made me wonder if it was something related to this recreation of the `SiteImageHandler`.